### ISSUE TITLE
Having \? in aresource file for Android project

### ISSUE THREAD

**Issue by razor313** [WeblateOrg/weblate#9570]
### Describe the issue

Once we receive/download all translations in the Android project there's \ along with a question mark like this `{MY_TEXT}\?`
But in the Weblate console, we don't have it! `{MY_TEXT}?`

I guess it has to do with this [commit](https://github.com/translate/translate/commit/60762a01565262e2463db60e7c38b42f3189646e).

Also, I've read this thread but didn't help me.

### I already tried

- [X] I've read and searched [the documentation](https://docs.weblate.org/).
- [X] I've searched for similar issues in this repository.

### Steps to reproduce the behavior

Define a new translation with `?`
Go to the file and download it, you will see it like `\?`

### How do you run Weblate?

Docker container

**Comment by nijel** [comment]
That's correct, the `?` should escaped in the file, see https://developer.android.com/guide/topics/resources/string-resource#escaping_quotes

**Comment by razor313** [comment]
@nijel Yes, that's right! Thank you for sharing. 🙏

**Comment by tipa** [comment]
I just noticed those escaped question marks in my downloads as well and was about to create an issue when I found this one.

 I had special characters in my XMLs for years (@, ?, chinese characters) without issues. Only characters that need escaping are `' " < > &`. When creating a new string resource in Android Studio, it does escape those, but no other characters - @, ? or chinese characters are **not** escaped.

So I am surprised about that [linked documentation](https://github.com/WeblateOrg/weblate/issues/9570#issuecomment-1635904484). And if Weblate aims to follow that documentation, wouldn't Chinese characters also have to be escaped? Because currently they aren't. 

Also, the Android docs say `you must escape the characters according to the standard XML/HTML escaping rules`.
However, whatever resource I pull up in the web, it only lists the 5 special characters I mentioned above:
https://www.liquid-technologies.com/Reference/Glossary/XML_EscapingData.html
https://stackoverflow.com/questions/1091945
https://docs.oracle.com/cd/A97335_02/apps.102/bc4j/developing_bc_projects/obcCustomXml.htm

So I don't know what "standard" Google is referring to here, but the entire internet seems to follow a different standard (as well as Android Studio - their own product).

I would appreciate if this bug report / change request could be reconsidered.
